// angular-stormpath/dist/esm/src/resend-email-verification/resend-email-verification.component.js
import { Component, Injectable, OnInit, Input, TemplateRef } from '@angular/core';
import { Stormpath, ResendEmailVerificationRequest } from '../stormpath/stormpath.service';
var ResendEmailVerificationComponent = /** @class */ (function () {
    function ResendEmailVerificationComponent(stormpath) {
        this.stormpath = stormpath;
    }
    ResendEmailVerificationComponent.prototype.ngOnInit = function () {
        this.posting = false;
        this.sent = false;
        this.formModel = {
            login: ''
        };
    };
    ResendEmailVerificationComponent.prototype.send = function () {
        var _this = this;
        this.posting = true;
        this.stormpath.resendVerificationEmail(this.formModel)
            .subscribe(function () {
            _this.posting = false;
            _this.sent = true;
        }, function (error) {
            _this.posting = false;
            _this.error = error.message;
        });
    };
    ResendEmailVerificationComponent.prototype.onSubmit = function () {
        this.send();
    };
    return ResendEmailVerificationComponent;
}());
export { ResendEmailVerificationComponent };
