// angular-stormpath/dist/esm/src/reset-password/reset-password.component.js
import { Component, Injectable, OnInit, TemplateRef, Input } from '@angular/core';
import { Location } from '@angular/common';
import { Stormpath, PasswordResetRequest, defaultSpTokenResolver } from '../stormpath/stormpath.service';
var ResetPasswordComponent = /** @class */ (function () {
    function ResetPasswordComponent(stormpath, location) {
        this.stormpath = stormpath;
        this.location = location;
    }
    ResetPasswordComponent.prototype.ngOnInit = function () {
        this.verifying = false;
        this.verified = false;
        this.verificationFailed = false;
        this.formModel = {
            sptoken: this.spTokenResolver(),
            password: ''
        };
        // debugger
        if (this.formModel.sptoken) {
            this.verify();
        }
    };
    ResetPasswordComponent.prototype.spTokenResolver = function () {
        return defaultSpTokenResolver(this.location);
    };
    ResetPasswordComponent.prototype.verify = function () {
        var _this = this;
        this.verifying = true;
        this.stormpath.verifyPasswordResetToken(this.formModel.sptoken)
            .subscribe(function () {
            _this.verifying = false;
            _this.verified = true;
        }, function (error) {
            _this.verifying = false;
            if (error.status && error.status === 404) {
                _this.verificationFailed = true;
            }
            else {
                _this.error = error.message;
            }
        });
    };
    ResetPasswordComponent.prototype.send = function () {
        var _this = this;
        this.stormpath.resetPassword(this.formModel)
            .subscribe(function () {
            _this.posting = false;
            _this.reset = true;
        }, function (error) {
            _this.posting = false;
            _this.error = error.message;
        });
    };
    ResetPasswordComponent.prototype.onSubmit = function () {
        this.send();
    };
    return ResetPasswordComponent;
}());
export { ResetPasswordComponent };
