// angular-stormpath/dist/esm/src/email-verification/email-verification.component.js
import { Component, OnInit, TemplateRef, Input } from '@angular/core';
import { Injectable } from '@angular/core';
import { Location } from '@angular/common';
import { Stormpath, defaultSpTokenResolver } from '../stormpath/stormpath.service';
var EmailVerificationComponent = /** @class */ (function () {
    function EmailVerificationComponent(stormpath, location) {
        this.stormpath = stormpath;
        this.location = location;
    }
    EmailVerificationComponent.prototype.ngOnInit = function () {
        this.verifying = false;
        this.verified = false;
        this.verificationFailed = false;
        this.sptoken = this.spTokenResolver();
        if (this.sptoken) {
            this.verify();
        }
    };
    EmailVerificationComponent.prototype.spTokenResolver = function () {
        return defaultSpTokenResolver(this.location);
    };
    EmailVerificationComponent.prototype.verify = function () {
        var _this = this;
        this.verifying = true;
        this.stormpath.verifyEmailVerificationToken(this.sptoken)
            .subscribe(function () {
            _this.verifying = false;
            _this.verified = true;
        }, function (error) {
            _this.verifying = false;
            if (error.status && error.status === 404) {
                _this.verificationFailed = true;
            }
            else {
                _this.error = error.message;
            }
        });
    };
    return EmailVerificationComponent;
}());
export { EmailVerificationComponent };
