// angular-stormpath/dist/esm/src/login/login.component.js
import { Component, OnInit, TemplateRef, Input } from '@angular/core';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { Account } from '../shared/account';
import { Stormpath, LoginFormModel, LoginService, StormpathErrorResponse } from '../stormpath/stormpath.service';
var LoginComponent = /** @class */ (function () {
    function LoginComponent(stormpath, loginService) {
        this.stormpath = stormpath;
        this.loginService = loginService;
    }
    LoginComponent.prototype.ngOnInit = function () {
        this.user$ = this.stormpath.user$;
        this.loggedIn$ = this.user$.map(function (user) { return !!user; });
        this.loginFormModel = {
            login: '',
            password: ''
        };
    };
    LoginComponent.prototype.login = function (form) {
        var _this = this;
        this.error = null;
        this.stormpath.login(this.loginFormModel)
            .subscribe(null, function (error) {
            _this.error = error.message;
        });
    };
    LoginComponent.prototype.forgot = function () {
        this.loginService.forgotPassword();
    };
    return LoginComponent;
}());
export { LoginComponent };
