// angular-stormpath/dist/esm/src/forgot-password/forgot-password.component.js
import { Component, OnInit, TemplateRef, Input } from '@angular/core';
import { Injectable } from '@angular/core';
import { Stormpath, ForgotPasswordFormModel, StormpathErrorResponse } from '../stormpath/stormpath.service';
var ForgotPasswordComponent = /** @class */ (function () {
    function ForgotPasswordComponent(stormpath) {
        this.stormpath = stormpath;
        this.sent = false;
    }
    ForgotPasswordComponent.prototype.ngOnInit = function () {
        this.forgotPasswordFormModel = {
            email: ''
        };
    };
    ForgotPasswordComponent.prototype.send = function () {
        var _this = this;
        this.error = null;
        this.stormpath.sendPasswordResetEmail(this.forgotPasswordFormModel)
            .subscribe(function () { return _this.sent = true; }, function (error) { return _this.error = error.message; });
    };
    ForgotPasswordComponent.prototype.onSubmit = function (form) {
        this.send();
    };
    return ForgotPasswordComponent;
}());
export { ForgotPasswordComponent };
