// angular-stormpath/dist/esm/src/authport/authport.component.js
import { Component, OnInit, TemplateRef, Input } from '@angular/core';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { Stormpath, LoginService } from '../stormpath/stormpath.service';
import { Account } from '../shared/account';
var AuthPortComponent = /** @class */ (function () {
    function AuthPortComponent(stormpath, loginService) {
        this.stormpath = stormpath;
        this.loginService = loginService;
        this.user$ = this.stormpath.user$;
        this.loggedIn$ = this.user$.map(function (user) { return !!user; });
    }
    AuthPortComponent.prototype.ngOnInit = function () {
        this.loginService.login = true;
        this.loginService.register = false;
        this.forgot = this.loginService.forgot;
        this.user$ = this.stormpath.user$;
        this.loggedIn$ = this.user$.map(function (user) { return !!user; });
    };
    AuthPortComponent.prototype.showLogin = function () {
        this.loginService.login = !(this.loginService.forgot = this.loginService.register = false);
    };
    AuthPortComponent.prototype.showRegister = function () {
        this.loginService.forgot = this.loginService.login = false;
        this.loginService.register = true;
    };
    AuthPortComponent.prototype.forgotPassword = function () {
        this.loginService.login = false;
        this.forgot = true;
    };
    AuthPortComponent.prototype.logout = function () {
        this.stormpath.logout();
    };
    return AuthPortComponent;
}());
export { AuthPortComponent };
